### PR TITLE
[GPU] Skip frequent sporadic in KVCacheTests

### PR DESCRIPTION
**Pull request by akladiev**
### Details:
There are many reproductions of the issue recently

### Tickets:
 - 142899

### CODE REVIEW

**Comment by akladiev** [comment]
Looks like test case is still not skipped for some reason...

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/skip_tests_config.cpp [hunk 1]
```diff
@@ -206,6 +206,8 @@ std::vector<std::string> disabledTestPatterns() {
             // by calc abs_threshold with expected value
             R"(.*smoke_RemoteTensor/OVRemoteTensorBatched_Test.NV12toBGR_buffer/(num_batch_4|num_batch_2).*)",
             R"(.*smoke_Check/ConstantResultSubgraphTest.Inference/SubgraphType=SINGLE_COMPONENT_IS=\[1,3,10,10\]_IT=i16_Device=GPU.*)",
+            // TODO: Issue: 142899
+            R"(.*smoke_multipleIterations_stateful_gather_with_initializer.*)",
```
Comment (yeonbok) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/skip_tests_config.cpp:210]: Could you please limit to the cases of "batch_1_5" and "batch_1_3" ? 
I guess that other tests have been existing more longer, so they would be okay.

Comment (yeonbok) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/skip_tests_config.cpp:210]: Also this is being skipped only for windows

Comment (akladiev) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/skip_tests_config.cpp:210]: Yes, it's reproducing on Windows only, that's why I'm trying to skip it here
But anyway I see that test case is still not skipped even on Windows ([logs](https://openvino-ci.toolbox.iotg.sclab.intel.com/blue/rest/organizations/jenkins/pipelines/private-ci/pipelines/ie/pipelines/ie-tests-windows-gpu/runs/105356/nodes/399/steps/404/log/?start=0)), maybe you have an idea why?

Comment (yeonbok) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/skip_tests_config.cpp:210]: I think to skip the test, we'll need SKIP_IF_CURRENT_TEST_IS_DISABLED() in the corresponding test in src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache.cpp

**Comment by yeonbok** [comment]
https://github.com/openvinotoolkit/openvino/pull/25043/files
Created a real fix to resolve the original issue, so no need to skip those test. Thanks!

**Pull request closed without merging**